esbuilder/lit-connect-modal/src/modal.js

```diff
@@ -0,0 +1,163 @@
+import MicroModal from 'micromodal';
+import css from './modal.css';
+import {
+  rawListOfWalletsArray,
+  metaMaskSingle,
+  coinbaseSingle,
+} from './helpers/walletList.js';
+import providerMethods from './helpers/providerMethods.js';
+
+export default class LitConnectModal {
+  constructor({ providerOptions }) {
+    this.dialog = MicroModal;
+    this.closeAction = undefined;
+    this.parent = document.body;
+    this.filteredListOfWalletsArray = [];
+    this.providerOptions = providerOptions;
+    this._filterListOfWallets();
+    this._instantiateLitConnectModal();
+
+    // inject css
+    var style = document.createElement('style');
+    style.innerHTML = css;
+    document.head.appendChild(style);
+  }
+
+  getWalletProvider() {
+    const currentProvider = localStorage.getItem('lit-web3-provider');
+
+    this.dialog.show('lit-connect-modal');
+    return new Promise((resolve, reject) => {
+      // if there is a current provider, resolve with it
+      if (!!currentProvider) {
+        const foundProvider = this.filteredListOfWalletsArray.find(
+          (w) => w.id === currentProvider
+        );
+        resolve(foundProvider.provider);
+        this._destroy();
+        return;
+      }
+
+      // otherwise, show the list of providers
+      this.filteredListOfWalletsArray.forEach((w) => {
+        let walletEntry = document.getElementById(w.id);
+        walletEntry.addEventListener('click', () => {
+          localStorage.setItem('lit-web3-provider', w.id);
+          resolve(w.provider);
+          this._destroy();
+          return;
+        });
+      });
+
+      this.closeAction.addEventListener('click', () => {
+        resolve(false);
+        this._destroy();
+        return;
+      });
+    });
+  }
+
+  _filterListOfWallets() {
+    const filteredListOfWalletsArray = [];
+
+    // -- this would only work if user installed multiple wallet extensions
+    // eg. when "ethereum.providers" is defined
+    rawListOfWalletsArray.forEach((w) => {
+      if (!!w['checkIfPresent'] && w['checkIfPresent']() === true) {
+        filteredListOfWalletsArray.push(w);
+      }
+    });
+
+    // -- try again, when user only installed a single wallet extension
+    // eg. when "ethereums.provider" it undefined, and can only be access
+    // via "ethereum.isMetaMask" or "ethereum.isCoinbaseWallet
+    if (filteredListOfWalletsArray.length === 0) {
+      if (globalThis.ethereum) {
+        if (globalThis.ethereum.isMetaMask) {
+          filteredListOfWalletsArray.push(metaMaskSingle);
+        }
+
+        if (globalThis.ethereum.isCoinbaseWallet) {
+          filteredListOfWalletsArray.push(coinbaseSingle);
+        }
+      }
+    }
+
+    // -- if walletconnect is present, add it to the list
+    if (!!this.providerOptions['walletconnect']) {
```

Comment:
is there a reason we need to key into the object instead of using standard '.' notation for property access here?